crates/bevy_ecs/src/query/iter.rs

```diff
@@ -475,6 +463,18 @@ impl<'w, 's, Q: WorldQuery, F: WorldQuery> QueryIterationCursor<'w, 's, Q, F> {
         }
     }
 
+    /// How many values will this cursor return?
+    fn remaining(&self, tables: &'w Tables, archetypes: &'w Archetypes) -> usize {
+        let remaining_matched: usize = if Self::IS_DENSE {
+            let ids = self.table_id_iter.clone();
+            ids.map(|id| tables[*id].len()).sum()
+        } else {
+            let ids = self.archetype_id_iter.clone();
+            ids.map(|id| archetypes[*id].len()).sum()
+        };
```

Comment:
this is gonna be wrong for stuff that isnt `Self::IS_ARCHETYPAL` i.e. `Query<(), Changed<T>>` would give the same hint as `Query<(), With<T>>` afaict ?

Reply:
Yes. This is true. Would adding that as a disclaimer to the method's doc be OK?